src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,193 @@
+            'If `perpendicular`, the categories can only move along a line perpendicular to the paths.',
+            'If `freeform`, the categories can freely move on the plane.',
+            'If `fixed`, the categories and dimensions are stationary.'
+        ].join(' ')
+    },
+    bundlecolors: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Sort paths so that like colors are bundled together within each category.'
+    },
+    sortpaths: {
+        valType: 'enumerated',
+        values: ['forward', 'backward'],
+        dflt: 'forward',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the path sorting algorithm.',
+            'If `forward`, sort paths based on dimension categories from left to right.',
+            'If `backward`, sort paths based on dimensions categories from right to left.'
+        ].join(' ')
+    },
+    labelfont: fontAttrs({
+        editType: 'calc',
+        description: 'Sets the font for the `dimension` labels.'
+    }),
+
+    categorylabelfont: fontAttrs({
+        editType: 'calc',
+        description: 'Sets the font for the `category` labels.'
+    }),
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        label: {
+            valType: 'string',
+            role: 'info',
+            editType: 'calc',
+            description: 'The shown name of the dimension.'
+        },
+        categoryorder: {
+            valType: 'enumerated',
+            values: [
+                'trace', 'category ascending', 'category descending', 'array'
+            ],
+            dflt: 'trace',
+            role: 'info',
+            editType: 'calc',
+            description: [
+                'Specifies the ordering logic for the categories in the dimension.',
+                'By default, plotly uses *trace*, which specifies the order that is present in the data supplied.',
+                'Set `categoryorder` to *category ascending* or *category descending* if order should be determined by',
+                'the alphanumerical order of the category names.',
+                'Set `categoryorder` to *array* to derive the ordering from the attribute `categoryarray`. If a category',
+                'is not found in the `categoryarray` array, the sorting behavior for that attribute will be identical to',
+                'the *trace* mode. The unspecified categories will follow the categories in `categoryarray`.'
+            ].join(' ')
+        },
+        categoryarray: {
+            valType: 'data_array',
+            role: 'info',
+            editType: 'calc',
+            description: [
+                'Sets the order in which categories in this dimension appear.',
+                'Only has an effect if `categoryorder` is set to *array*.',
+                'Used with `categoryorder`.'
+            ].join(' ')
+        },
+        categorylabels: {
```

Comment:
Would it be a stretch to call this `ticktext` and `categorylabelfont` -> `tickfont` to preserve more "symmetry" with other trace types?

Reply:
Ok, after playing with `categoryorder`, `categoryarray`, `ticktext`, and `tickvals` in `layout.xaxis` I think I do see the parallel.

What I'm calling `categorylabels` would be equivalent to `ticktext`  if we were to introduce a `tickvals` property and always keep it equal to `categoryarray`.  I don't want to actually introduce a `tickvals` property (because I don't see how it would be useful in general), but if we think of it as being there and always fixed to be equal to `categoryarray` then I think the parallel holds.

So I think I'm fine with these renames, any thoughts @alexcjohnson?

Reply:
Done in https://github.com/plotly/plotly.js/pull/2963/commits/5a19499b5fe56f4f50c44939c4d2784491d356f7!